### PR TITLE
[OC-1218] Frontend : Create UI which allows the user to create their …

### PR DESCRIPTION
**Pull request by vlo-rte**
The jira ticket is [OC-1218](https://opfab.atlassian.net/browse/OC-1218).
Frederic, I propose for release notes : 

[OC-1170] Filter notification screen for user (Work still in progress). Steps done : Backend filter storage and creation of the UI

[OC-1170]: https://opfab.atlassian.net/browse/OC-1170

### CODE REVIEW

File: ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.ts [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+                                                            { stateLabel: string,
+                                                              stateControlIndex: number
+                                                            }[]
+                                                       }> ();
+        this.preparedListOfProcessesStates = [];
+        this.processesWithoutGroup = [];
+
+        this.computePreparedListOfProcessesStatesAndProcessesStatesLabels();
+        this.makeProcessesWithoutGroup();
+        this.addCheckboxesInFormArray();
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.ts:89]: this.addCheckboxesInFormArray() should be call i guess in the ngOnInit method . There is actually a bug when you change screen and return after to this screen , the checkbox selected and saved are not checked anymore

File: ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.ts [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+
+class Message {
+    text: string;
+    display: boolean;
+}
+
+function getI18nPrefixFromProcess(process: Process): string {
+    return process.id + '.' + process.version + '.';
+}
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.ts:32]: Put inside the FeedconfigurationComponent

File: ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.ts [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+                                     stateId: string }[];
+
+    modalRef: NgbModalRef;
+
+    public displaySendResultOk = false;
+    public displaySendResultError = false;
+    messageAfterSavingSettings: string;
+
+    unsubscribe$: Subject<void> = new Subject<void>();
+
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.ts:63]: Not used --> to remove

**Review by freddidierRTE — CHANGES_REQUESTED** [review]

File: ui/main/src/app/app.component.ts [hunk 4]
```diff
@@ -119,6 +119,7 @@ export class AppComponent implements OnInit {
             this.userService.loadUserWithPerimetersData(),
             this.entitiesService.loadAllEntitiesData(),
             this.processesService.loadAllProcesses(),
+            this.processesService.loadProcessGroups(),
             this.processesService.areTranslationsLoaded(),
             this.cardService.initSubscription)
             .pipe(skip(4)) // Need to wait for all initialization to complete before loading main components of the application
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/app.component.ts:125]: Done

File: ui/main/src/app/services/processes.service.ts [hunk 5]
```diff
@@ -88,6 +91,22 @@ export class ProcessesService {
...
+                        if (!!processGroupsFile) {
+                            this.processGroups = processGroupsFile.groups;
+
+                            for (const language in processGroupsFile.locale)
+                                 this.translateService.setTranslation(language, processGroupsFile.locale[language], true);
+
+                            console.log(new Date().toISOString(), 'List of process groups loaded');
+                        }
+                    }, (error) => console.error(new Date().toISOString(), 'an error occurred', error)
+                ));
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/services/processes.service.ts:107]: Done

File: ui/main/src/app/modules/settings/settings.module.ts [hunk 6]
```diff
@@ -31,6 +31,9 @@ import { CheckboxSettingComponent } from './components/settings/checkbox-setting
         , MultiSettingsComponent
         , TypeAheadSettingsComponent
         , CheckboxSettingComponent],
+    exports: [
+        CheckboxSettingComponent
+    ],
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/settings/settings.module.ts:36]: Done

File: ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.ts [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+        }
+    }
+
+    ngOnInit() {
+    }
+
+    ngOnDestroy() {
+        this.unsubscribe$.next();
+        this.unsubscribe$.complete();
+    }
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.ts:169]: Done. Removed.

File: ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.scss [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+.modal-body{
+  background-color: var(--opfab-lightcard-detail-bgcolor);
+  color: var(--opfab-lightcard-detail-textcolor);
+}
+
+
+.modal-header{
+  background-color: var(--opfab-lightcard-detail-bgcolor);
+  color: var(--opfab-lightcard-detail-textcolor);
+}
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.scss:19]: Done.

File: ui/main/src/app/modules/feedconfiguration/feedconfiguration.module.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+@NgModule({
+    declarations: [FeedconfigurationComponent],
+    imports: [
+        CommonModule,
+        FormsModule,
+        ReactiveFormsModule,
+        TranslateModule,
+        FeedconfigurationRoutingModule,
+        SettingsModule
+    ]
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/modules/feedconfiguration/feedconfiguration.module.ts:28]: Done. Import removed.

File: ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.ts (hunk 2 above)
Comment (vlo-rte) [reply ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.ts:32]: Done.

File: ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.ts (hunk 3 above)
Comment (vlo-rte) [reply ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.ts:63]: Done.

File: ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.ts (hunk 1 above)
Comment (vlo-rte) [reply ui/main/src/app/modules/feedconfiguration/feedconfiguration.component.ts:89]: Done.

File: ui/main/src/app/components/navbar/navbar.component.html [hunk 10]
```diff
@@ -102,6 +102,16 @@
...
+
+          <div class="dropdown-item" [class.active]="currentPath && 'feedconfiguration' === currentPath[1]">
+            <a
+                    class="nav-link opfab-menu-link text-decoration-none"
+                    routerLink="/feedconfiguration"
+                    routerLinkActive #rla="routerLinkActive"
+                    translate
+            >menu.feedConfiguration</a>
+          </div>
+
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/app/components/navbar/navbar.component.html]: Done

**Pull request merged**